### ISSUE TITLE
Enhance the implementation for k range frame of window operator

### ISSUE THREAD

**Issue by PHILO-HE** [facebookincubator/velox#7557]
### Description

There are already some discussions before. cc @aditi-pandit
* Support constant start/end value (preceding/following offset relative to current row value) and let velox compute the lower/upper bound value:
https://github.com/facebookincubator/velox/pull/7303#discussion_r1390740185
* Frame bound search optimization:
https://github.com/facebookincubator/velox/pull/7303#discussion_r1392033603
* Support types like date, timestamp, interval, etc.

**Comment by mbasmanova** [comment]
@PHILO-HE Would you consider using ProjectNode before WindowNode to precompute these values?

**Comment by PHILO-HE** [comment]
> @PHILO-HE Would you consider using ProjectNode before WindowNode to precompute these values?

@mbasmanova, yes, it is better to directly construct a ProjectNode before WindowNode. I will try to do that after @aditi-pandit's patch is merged. Thanks for your suggestion!
